src/digital_ocean.rs

```diff
@@ -74,19 +74,38 @@ impl DigitalOceanClient {
                 let ip_address_array = droplet_json["networks"]["v4"].as_array().ok_or(
                     Error::MalformedDigitalOceanApiRespose("droplets".to_string()),
                 )?;
-                let public_ip = ip_address_array
-                    .iter()
-                    .find(|x| x["type"].as_str().unwrap() == "public")
-                    .ok_or(Error::DigitalOceanPublicIpAddressNotFound)?;
-                droplets.push(Droplet {
-                    id: id as usize,
-                    name,
-                    ip_address: Ipv4Addr::from_str(
+                // The following might fail if we start multiple networks in parallel.
+                let get_ip_address = || -> Result<Ipv4Addr, Error> {
+                    let public_ip = ip_address_array
+                        .iter()
+                        .find(|x| x["type"].as_str().unwrap() == "public")
+                        .ok_or(Error::DigitalOceanPublicIpAddressNotFound)?;
+
+                    let ip_address = Ipv4Addr::from_str(
                         public_ip["ip_address"]
                             .as_str()
                             .ok_or(Error::DigitalOceanPublicIpAddressNotFound)?,
-                    )?,
-                });
+                    )?;
+
+                    Ok(ip_address)
+                };
+
+                match get_ip_address() {
+                    Ok(ip_address) => {
+                        droplets.push(Droplet {
+                            id: id as usize,
+                            name,
+                            ip_address,
+                        });
+                    }
+                    Err(err) => {
+                        if skip_if_no_ip {
+                            continue;
+                        } else {
+                            return Err(err);
+                        }
+                    }
```

Comment:
Just a suggestion, not sure if it's neater.

```suggestion
                    Err(_) if skip_if_no_ip => continue,
                    Err(err) => return Err(err),
```

Reply:
This is much neater! 👍